src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,51 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int NUMBER_LIMIT = 100;
+    private static final int defaultRadius = 10;
+    private Random random = new Random();
+
+    public Figure getRandomFigure() {
+        ColorSupplier colorSupplier = new ColorSupplier();
```

Comment:
ColorSupplier object creates every time you call getRandomFigure/getDefaultFigure. Think of approach, when you create an object of ColorSupplier only once for this class